### PR TITLE
Send start and completion time to external logs proxy

### PR DESCRIPTION
**Pull request by AverageMarcus**
# Changes

Fixes https://github.com/tektoncd/dashboard/issues/2908

Include the `startTime` and `completionTime` of the TaskRun when calling out to the external logs proxy so that queries against the long-term log storage can be optimised to search a targeted timeframe if possible.

These are included as query parameters to allow for backwards-compatibility with existing route handling logic that might be being used. 

/kind feature

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [ ] [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
	- No existing docs for external-logs aside from the walkthrough. This change doesn't cause a change to those docs.
- [x] [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [x] Release notes block below has been updated with any user facing changes (new features, significant UI changes, API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [x] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
Include the TaskRun's `startTime` and `completionTime` as query parameters when making requests to the external-logs proxy.
```

### CODE REVIEW

File: src/api/index.js [hunk 1]
```diff
@@ -185,9 +185,19 @@ export function getExternalLogURL({
...
 }) {
-  return `${externalLogsURL}/${namespace}/${podName}/${container}`;
+  let queryStringParams = []
+  if (startTime) queryStringParams.push(`startTime=${startTime}`)
+  if (completionTime) queryStringParams.push(`completionTime=${completionTime}`)
+
+  let queryString = ""
+  if (queryStringParams.length) {
+    queryString = `?${queryStringParams.join('&')}`
+  }
```
Comment (AlanGreene) [inline src/api/index.js:199]: Can you update this to use `URLSearchParams` instead? This ensures that if in future we add values that need special encoding they'll be properly handled.

e.g.

```js
const queryParams = new URLSearchParams();
if (startTime) {
  queryParams.set('startTime', startTime);
}
if (completionTime) {
  queryParams.set('completionTime', completionTime);
}
let queryString = queryParams.toString(); // returns the properly encoded string, or '' if no params
if (queryString) {
  queryString = `?${queryString}`;
}

return `${externalLogsURL}/${namespace}/${podName}/${container}${queryString}`
```

**Review by AlanGreene — CHANGES_REQUESTED** [review]
Hi @AverageMarcus, thanks for the contribution (including tests!). Overall this approach looks good, should be backwards compatible, and could be useful to other users, so it makes sense to include it 👍 

I'm made a few small suggestions below, let me know if you have any questions.

File: src/api/index.js [hunk 2]
```diff
@@ -185,9 +185,19 @@ export function getExternalLogURL({
   container,
   externalLogsURL,
   namespace,
-  podName
+  podName,
+  startTime,
+  completionTime
```
[the remark this answers is not in the document]
Comment (AverageMarcus) [reply src/api/index.js]: Done!

File: src/api/index.js (hunk 1 above)
Comment (AverageMarcus) [reply src/api/index.js]: Great suggestion! Updated!

**Review by AlanGreene — APPROVED** [review]
Thanks Marcus
/lgtm

**Pull request merged**